exercicios/para-casa/sistema_de_alunas.py

```diff
@@ -1,59 +1,168 @@
+        while True:  # não sabemos a quantidade de repetições
+            entrada = input(f'⋆.˚ Insira a nota #{i+1}: ')
+            try:
+                nota = (round(float(entrada), 2))
+                notas.append(nota)
+                break  # sai do while e segue com for
+            except ValueError:
+                print("❌🤖 Entrada inválida. Insira um número válido.")
+
+    print(f'🧸ྀི  Confirmando as notas recebidas: {notas} 📂')
+    return notas
+
+
+def presencas():
+    ata = []
+
+    for i in range(5):
+        while True:
+            entrada = input(f'⋆.˚ Digite a presença da aluna do dia {i+1} (true/false): ').strip().lower()
+            if entrada == 'true':
+                presenca_dia = True
+                ata.append(presenca_dia)
+                break
+            elif entrada == 'false':
+                presenca_dia = False
+                ata.append(presenca_dia)
+                break
+            else:
+                print("❌🤖 Entrada inválida. Por favor, insira True ou False.")
+
+    print(f'🧸ྀི  Confirmando as presenças recebidas: {ata} 📂')
+    return ata
+
+
 def consultar_lista_alunas():
-    pass
-    #TODO - Implentar a função
-    
+    print("🪼⋆.ೃ࿔*:･ Lista de alunas:")
+    for (nome, sobrenome), info in dataset.items():
+        pprint.pprint(f"Nome: {nome} {sobrenome}, Turma: {info['Turma']}")
+
+
 def consultar_faltas_aluna():
-    pass
-    #TODO - Implentar a função
-    
+    nome = input("🫧 Digite o primeiro nome da aluna: ").capitalize()
+    sobrenome = input("🫧 Digite o sobrenome da aluna: ").strip().capitalize()
+    chave = (nome, sobrenome)
+    if chave in dataset:
+        presenca = dataset[chave]["Presença"]
+        faltas = presenca.count(False)
+        print(f"🪼⋆. A quantidade de faltas da aluna {nome} {sobrenome} é de {faltas} faltas. 🪼⋆.")
+    else:
+        print("❌ Aluna não foi encontrada.")
+
+
 def consultar_notas_aluna():
-    pass
-    #TODO - Implentar a função
-    
+    nome = input("🫧 Digite o primeiro nome da aluna: ").capitalize()
+    sobrenome = input("🫧 Digite o sobrenome da aluna: ").strip().capitalize()
+    chave = (nome, sobrenome)
+    if chave in dataset:
+        notas = dataset[chave]["Notas"]
+        print(f"🪼⋆. Notas da aluna {nome} {sobrenome}: {notas}  🪼⋆.")
+    else:
+        print("❌ Aluna não foi encontrada.")
+
+
 def consultar_status_aprovacao():
-    pass
-    #TODO - Implentar a função
-    
+    nome = input("🫧 Digite o primeiro nome da aluna: ").capitalize()
+    sobrenome = input("🫧 Digite o sobrenome da aluna: ").strip().capitalize()
+    chave = (nome, sobrenome)
+    if chave in dataset:
+        notas = dataset[chave]["Notas"]
+        presenca = dataset[chave]["Presença"]
+        participacao = dataset[chave]["Participação"]
+
+        media = sum(notas) / len(notas)
+        num_presenca = presenca.count(True)
+
+        if media >= 6 and participacao >= 6 and num_presenca >= 0.8*len(presenca):
```

Comment:
Evite colocar o cálculo direto na condicional. Para tornar o código mais limpo, crie uma variável e compare posteriormente.

**Sugestão:**
```suggestion
        percentual_presenca = (num_presenca / len(presenca))
        if media >= 6 and participacao >= 6 and percentual_presenca >= 0.8:
```